tests/integration/cli/src/object_file_engine_test_c_source.c

```diff
@@ -0,0 +1,145 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "wasmer_wasm.h"
+#include "wasm.h"
+#include "my_wasm.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+void wasmer_function__1(void);
+void wasmer_trampoline_function_call__1(void*, void*, void*);
+
+#ifdef __cplusplus
+}
+#endif
+
+void print_wasmer_error()
+{
+    int error_len = wasmer_last_error_length();
+    printf("Error len: `%d`\n", error_len);
+    char* error_str = (char*) malloc(error_len);
+    wasmer_last_error_message(error_str, error_len);
+    printf("Error str: `%s`\n", error_str);
+}
+
+
+int main() {
+        printf("Initializing...\n");
+        wasm_config_t* config = wasm_config_new();
+        wasm_config_set_compiler(config, CRANELIFT);
+        wasm_config_set_engine(config, OBJECT_FILE);
```

Comment:
This seem a bit wrong. We should create a file object **headless** engine.

Reply:
Meaning: it should not require any compiler attached to Wasmer

Reply:
Yeah, currently whether or not an engine is headless is determined purely by whether the `libwasmer_c_api` was compiled with the `compiler` feature or not.

I agree that it'd be good to allow users to select it too.

Reply:
I mean that the example should not even mention a compiler here.

Can we just remove the line?
```
        wasm_config_set_compiler(config, CRANELIFT);
```

Reply:
In this kind of engine, the wasm is already precompiled.
So is not that users should be able to select it... is more that it doesn't make sense to select any compiler at all for this engine example.

Reply:
Oh sure, we can remove it but it's functionally the same: it will default to using `Cranelift`, I was just being explicit but yeah we can remove it.

Reply:
We can make it so that absence of a compiler means headless mode, but perhaps it's better to be explicit about it because most users will want to use a compiler most of the time

Reply:
Yeah, but not in the case of the object-file engine. In this case compilers are not needed nor desired